core/domain/platform_parameter_domain.py

```diff
@@ -911,34 +885,38 @@ def _validate_feature_flag(self) -> None:
                 'Data type of feature flags must be bool, got \'%s\' '
                 'instead.' % self._data_type)
         if not any(
-                self._feature_stage == feature_stage
-                for feature_stage in ALLOWED_FEATURE_STAGES
+            self._feature_stage == feature_stage
+            for feature_stage in ALLOWED_FEATURE_STAGES
         ):
             raise utils.ValidationError(
                 'Invalid feature stage, got \'%s\', expected one of %s.' % (
                     self._feature_stage, ALLOWED_FEATURE_STAGES))
-        enabling_rules = [
-            rule for rule in self._rules if rule.value_when_matched]
-        for rule in enabling_rules:
-            server_mode_filters = [
-                server_mode_filter for server_mode_filter in rule.filters
-                if server_mode_filter.type == 'server_mode']
-            for server_mode_filter in server_mode_filters:
-                server_modes = [
-                    value for _, value in server_mode_filter.conditions]
-                if self._feature_stage == FeatureStages.DEV.value:
-                    if (
-                            ServerMode.TEST.value in server_modes or
-                            ServerMode.PROD.value in server_modes
-                    ):
-                        raise utils.ValidationError(
-                            'Feature in dev stage cannot be enabled in test or'
-                            ' production environments.')
-                elif self._feature_stage == FeatureStages.TEST.value:
-                    if ServerMode.PROD.value in server_modes:
-                        raise utils.ValidationError(
-                            'Feature in test stage cannot be enabled in '
-                            'production environment.')
+
+        server_mode = (
+            ServerMode.DEV
+            if constants.DEV_MODE
+            else ServerMode.PROD
+            if feconf.ENV_IS_OPPIA_ORG_PRODUCTION_SERVER
+            else ServerMode.TEST
+        )
+        if (
+            server_mode.value == ServerMode.TEST.value and
+            self._feature_stage == ServerMode.DEV.value
+        ):
+            raise utils.ValidationError(
+                'Feature in %s stage cannot be enabled in %s environment.' % (
```

Comment:
What about this check suggests "enabled"? Is it more like it cannot be updated?

Ditto below.

Reply:
Updated.